source/fundamentals/crud/write-operations/insert.txt

```diff
@@ -1,12 +1,167 @@
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
 Overview
 --------
 
-In this section, we show you how to call the write operations to **insert**
-documents from a collection in your MongoDB database.
+In this guide, you will learn how to insert documents into MongoDB. 
+
+You can use MongoDB to retrieve, update and delete information. To
+perform any of those operations, that information, such as user profiles
+and orders, needs to exist in MongoDB. For that information to exist,
+you need to first perform an insert operation.
+
+An insert operation inserts a single or multiple documents in MongoDB
+using the  ``insertOne()``, ``insertMany()`` and ``bulkWrite()``
+methods. 
+
+The following sections focus on ``insertOne()`` and
+``insertMany()``. For information on how to use the ``bulkWrite()``
+method, see our ``Bulk Operations <TODO>`` page. 
+
+A Note About ``_id``
+--------------------
+
+When inserting a document, you have the choice to include certain fields
+and values except for the ``_id`` field. By default, MongoDB makes the
+``_id`` field our unique index.
+
+There are two ways to manage this field:
+
+- You can provide the same type for each ``_id``.
+- You can omit the field to let the driver automatically generate unique values.
+
+Unless you have another way to guarantee a unique index, let the
+driver to automatically generate ``_id`` values. 
+
+.. note::
+
+   If you violate the unique index constraint, the insertion fails and a
+   ``writeError`` occurs. 
+ 
+Insert a Single Document
+------------------------
+
+Use the ``insertOne()`` method when you want to insert a single
+document. 
+
+If the insertion is successful, the method returns an
+``InsertOneResult`` instance representing the ``_id`` of the new
+document. 
+
+Example
+~~~~~~~
+
+The following example creates and inserts a document using the
+``insertOne()`` method: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Insert.java
+   :language: java
+   :dedent:
+   :start-after: begin insertOneExample
+   :end-before: end insertOneExample
+
+Your output should look something like this:
+
+.. code-block:: json
+   :copyable: false
+
+   Inserted a document with the following id: 60930c39a982931c20ef6cd6
+
+Insert Multiple Documents
+-------------------------
+
+Use the ``insertMany()`` method when you want to insert multiple
+documents. This method inserts documents in the order specified until an
+exception occurs if any.
+
+For example, assume you want to insert the following documents:
+
+.. code-block:: json
+
+   { "_id": 3, "color": "red", "qty": 5 }
+   { "_id": 4, "color": "purple", "qty": 10 }
+   { "_id": 3, "color": "yellow", "qty": 3 }
+   { "_id": 6, "color": "blue", "qty": 8 }
+
+If you attempt to insert these documents, a ``writeError`` occurs and
+the documents where the ``color`` is "red"  and "purple" are in your
+collection. 
+
+.. warning::
+
+   There is no acknowledgment for the documents that inserted before
```

Comment:
```suggestion
   There is no acknowledgment for the documents that are inserted before
```

Reply:
did you verify this with Jeff/Ross?

Reply:
There's a thread pending their reply in the java channel, but I put this warning in for now more as a placeholder.  If there is an acknowledgement, I'll change this to a note with they way to do so.

Reply:
Sounds good